lib/mongo/retryable.rb

```diff
@@ -131,7 +131,7 @@ def read_with_retry(session = nil, server_selector = nil, &block)
         server = select_server(cluster, server_selector, session)
         begin
           yield server
-        rescue Error::SocketError, Error::SocketTimeoutError, Error::ServerNotUsable, Error::PoolError, Error::OperationFailure => e
+        rescue Error::SocketError, Error::SocketTimeoutError, Error::ServerNotUsable, Error::ConnectionPerished, Error::PoolError, Error::OperationFailure => e
```

Comment:
This looks really ugly now, maybe we should introduce something like`Error::Retryable` mixin, and catch it here.

Reply:
Let's file a followup ticket?

Reply:
RUBY-3191

Reply:
👍